README.md

```diff
@@ -43,17 +57,127 @@ Snippets for deployment template scaffolding and 70+ Azure resources.
 
-You can also right-click on any element in the ARM Template Outline view to initiate the insert item process.
+Entering double quotes `""` or selecting `ctrl` + `space` while the cursor is in a resource declaration provides a list of potential properties and property values for each Azure resource.
 
-![Inserting an item into an Azure Resource Manager template](./images/insert-item-outline.png)
+In the following example, the potential values for an Azure Storage account tier are returned from the Azure Storage schema.
+
+![Image showing a list of storage account tier values returned from the storage Azure schema](./images/completion.png)
+
+### Schema Validation
+
+If an invalid property or property value is specified, a validation warning is thrown. In the following example, an invalid value is provided for the storage account tier (line 19). Notice that the warning message includes a list of valid values.
+
+![Image showing a schema validation error](./images/schema-validation.png)
+
+## Parameter files
+
+Associating a parameter file with an ARM template enables full validation across both the parameter file and template. A parameter file association can be made by either creating a new parameter file from an existing template or by associating an existing parameter file to the template.
+
+### Create a new parameter file
+
+To create a new parameter file from an existing template, click on the code lense text `**Select or create a parameter file to enable full validation..**.
+
+![Image showing the code lense for creating a parameter file association](./images/create-parameter-file.png)
+
+Select **New** from the context menu.
+
+![Image showing the parameter file creation options (none, new, and browse)](./images/create-parameter-file-2.png)
+
+Selecting **All parameters** adds every parameter from the template into the new parameter file. Selecting **Only required parameters** adds only parameters that do not have a default value specified in the template file.
+
+![Image showing options for adding parameters for parameter file (All parameters, only required parameters)](./images/create-parameter-file-3.png)
+
+Select a name and location for the parameter file.
+
+![Image showing a file save as dialog box](./images/create-parameter-file-4.png)
+
+Once done, a parameter file has been created. Where needed, remove the comment indicating that a parameter value is needed and add a parameter value.
+
+![Image showing new parameter file](./images/parameter-file.png)
+
+Also, notice that on the template file that the code lense text has been updated to reflect the current parameter file association.
+
+![Image showing that the parameter code lens has been updated to reflect associated parameter file](./images/code-lense.png)
+
+### Associate an existing parameter file
+
+To create an association with an existing parameter file, click on the code lense text `**Select or create a parameter file to enable full validation..**.
```

Comment:
extra period ".." and missing the closing backtick

Reply:
Fixed